src/pages/analysis/AnalysisPage.tsx

```diff
@@ -1,55 +1,140 @@
-import { analysis } from '@/images';
-import { Colors, FontSize, FontWeight, Media } from '@/styles';
+import { useCallback, useEffect, useState } from 'react';
+import { Media } from '@/styles';
+import { routineStatusApi } from '@/apis';
+import { Container, Calendar, Routine, Spinner } from '@/components';
+import { useHistory } from 'react-router-dom';
+import { RoutineStatusType } from 'Models';
+
+interface highlightDatesType {
+  [key: string]: number;
+}
+
+const parseHighlightDates = (
+  routineStatusData: RoutineStatusType[],
+): highlightDatesType => {
+  const highlightDates: highlightDatesType = {};
+
+  routineStatusData.forEach((routineStatus: RoutineStatusType) => {
+    const date = dayjs(routineStatus.dateTime.slice(0, 11)).format(
+      'YYYY-MM-DD',
+    );
+    highlightDates[date]
+      ? (highlightDates[date] += 1)
+      : (highlightDates[date] = 1);
+  });
+
+  return highlightDates;
+};
 
 const AnalysisPage = (): JSX.Element => {
+  const history = useHistory();
+  const [loading, setLoading] = useState<boolean>(false);
+  const [highlightDates, setHighlightDates] = useState<highlightDatesType>({});
+  const [routineStatuses, setRoutineStatuses] = useState<RoutineStatusType[]>(
+    [],
+  );
+
+  const getRoutineStatusByDate = useCallback(
+    async (date: string): Promise<RoutineStatusType[]> => {
+      try {
+        const res = await routineStatusApi.getRoutineStatusByDate(date);
+        const routineStatus = res.data.data;
+        return routineStatus;
+      } catch (error) {
+        return [];
+      }
+    },
+    [],
+  );
+
+  const handleClickDate = async (date: dayjs.Dayjs) => {
+    setLoading(true);
+    const dateString = date.format('YYYY-MM-DD');
+    const routineStatuses = await getRoutineStatusByDate(dateString);
+    setRoutineStatuses(routineStatuses);
+    setLoading(false);
+  };
+
+  const handleChangeYearMonth = async (yearMonth: dayjs.Dayjs) => {
+    setLoading(true);
+    const yearMonthString = yearMonth.format('YYYY-MM');
+    const routineStatuses = await getRoutineStatusByDate(yearMonthString);
+    const highlightDates = parseHighlightDates(routineStatuses);
+    setHighlightDates(highlightDates);
+    setLoading(false);
+  };
+
+  const handleClickRoutine = (routineStatusId: number) => {
+    history.push(`/analysis/detail/${routineStatusId}`);
+  };
+
+  useEffect(() => {
+    const init = async () => {
+      setLoading(true);
+      const now = dayjs();
+      const yearMonth = now.format('YYYY-MM');
+      const todayDate = now.format('YYYY-MM-DD');
+      const yearMonthRoutineStatuses = await getRoutineStatusByDate(yearMonth);
+      const todayDateRoutineStatuses = await getRoutineStatusByDate(todayDate);
+      const highlightDates = parseHighlightDates(yearMonthRoutineStatuses);
+      setHighlightDates(highlightDates);
+      setRoutineStatuses(todayDateRoutineStatuses);
+      setLoading(false);
+    };
+
+    init();
+  }, [getRoutineStatusByDate]);
```

Comment:
`useEffect`의 종속성으로 들어간 `getRoutineStatusByDate` 함수가 `useCallBack`으로 정의돼있고, `useCallback`을 작성할 때 두 번째 인자로 빈 배열이 들어가면 최초 생성된 함수를 지속적으로 기억한다고 하네요!
그래서 함수 자체가 재정의되지 않아서 종속성으로 작성한 의미가 없어지고, `useEffect`를 Life Cycle처럼 사용할 것을 의도한 거라면 중요한 부분은 아닐 수도 있지만..😂 빈 배열로 바꿔도 동일하게 동작할 듯 합니다~!!

```suggestion
  }, []);
```

Reply:
오 준형님~ 나이스한 캐치네요!!! 👍

deps를 최대한 작성을 안하려고 하는데 아래와 같이 경고가 발생하여 추가해주었습니다. 😂

그치만 준형님이 말씀하신대로 다시 보니 왜 경고가 발생하는지는 저도 의문이네요 코드를 뜯어보면서 고민좀 해봐야겠습니다.

![2022-01-31_19-26-09](https://user-images.githubusercontent.com/41064875/151778581-ddc6fc8a-801f-4a55-bee2-179cd998b259.png)

Reply:
어떤 부분에서 `getRoutineStatusByDate` 함수가 변경될 가능성이 있어서 그런지 모르겠네요 ..흠...

오히려 useCallback을 삭제하니 deps를 제거해도 되네요...(?)

현재는 useCallback을 사용할 이유가 없다 생각이 들어 useCallback을 삭제하는식으로 리팩토링을 하였습니다.

그런데 정말 왜 이런 현상이 있는지 궁금한네요 🧐

Reply:
음 윤호님 useCallback 아예 삭제하고 useEffect 부분에 콘솔찍어보면 무한으로 계속 찍히나요..?
어렵네요ㅎㅎㅎ 목욜에 같이 이야기 나눠봐요~~!!!!

Reply:
useCallback없이 의존성에 추가해주면 무한 리렌더링됩니다 🤪